client/src/utils.ts

```diff
@@ -229,17 +229,18 @@ export const deepHtmlStringMatch = (element: any, match: string) => {
 
 // Given a HTML img element, process image data for barcode string
 export const processImageBarcode = async (img: HTMLImageElement) => {
-  const codeReader = new BrowserMultiFormatReader();
-  if (img) {
-    const result = await codeReader
-      .decodeFromImage(img)
-      .then((res: any) => res.text)
-      .catch((err: any) => null);
-    if (result) {
+  try {
+    const codeReader = new BrowserMultiFormatReader();
+    if (img) {
+      const result = await codeReader
+        .decodeFromImage(img)
+        .then((res: any) => res.text)
+        .catch((err: any) => null);
       return result;
     }
+  } catch (err) {
+    return null;
```

Comment:
Prefer to avoid nulls in code. Log error here & throw exception instead

Reply:
changed, `console.error` followed by throwing the error, added catch for this when used in `MediaScanner`